packages/gatsby-transformer-sharp/src/extend-node-type.js

```diff
@@ -57,11 +59,48 @@ const DuotoneGradientType = new GraphQLInputObjectType({
   },
 })
 
+const PotraceType = new GraphQLInputObjectType({
+  name: `Potrace`,
+  fields: () => {
+    return {
+      turnPolicy: {
+        type: new GraphQLEnumType({
+          name: `PotraceTurnPolicy`,
+          values: {
+            TURNPOLICY_BLACK: { value: Potrace.TURNPOLICY_BLACK },
+            TURNPOLICY_WHITE: { value: Potrace.TURNPOLICY_WHITE },
+            TURNPOLICY_LEFT: { value: Potrace.TURNPOLICY_LEFT },
+            TURNPOLICY_RIGHT: { value: Potrace.TURNPOLICY_RIGHT },
+            TURNPOLICY_MINORITY: { value: Potrace.TURNPOLICY_MINORITY },
+            TURNPOLICY_MAJORITY: { value: Potrace.TURNPOLICY_MAJORITY },
+          },
+        }),
+      },
+      turdSize: { type: GraphQLFloat },
+      alphaMax: { type: GraphQLFloat },
+      optCurve: { type: GraphQLBoolean },
+      optTolerance: { type: GraphQLFloat },
+      threshold: { type: GraphQLInt },
+      blackOnWhite: { type: GraphQLBoolean },
+      color: { type: GraphQLString },
+      background: { type: GraphQLString },
+    }
+  },
+})
+
 module.exports = ({ type, pathPrefix, getNodeAndSavePathDependency }) => {
   if (type.name !== `ImageSharp`) {
     return {}
   }
 
+  const getTracedSVG = parent => {
+    const promise = traceSVG({
+      file: { absolutePath: path.join(process.cwd(), `public`, parent.src) },
```

Comment:
So I just wondered if what I'm currently doing is actually close to what you suggested …
The reason I wondered about this is that I didn't know how to actually reach the original image path from the subfield, so I just used what was there, namely `parent.src`…

And after coincidentally deleting my `public` folder, it turns out that `parent.src` isn't necessarily available at this point in time, meaning all that I thought would work up to now really isn't ;-/ … so much for "you got this". 😿